pages/table-fragments/grid-navigation-custom.page.tsx

```diff
@@ -56,17 +64,19 @@ const actionsModeOptions = [
 export default function Page() {
   const [toolsOpen, setToolsOpen] = useState(false);
   const { urlParams, setUrlParams } = useContext(AppContext as PageContext);
-  const pageSize = urlParams.pageSize ?? 10;
-  const tableRole = urlParams.tableRole ?? 'grid';
-  const actionsMode = urlParams.actionsMode ?? 'dropdown';
+  const { pageSize = 10, tableRole = 'grid', actionsMode = 'dropdown', autoRefresh = false } = urlParams;
 
   const [items, setItems] = useState(generateItems(25));
+  const [refreshCounter, setRefreshCounter] = useState(0);
+  window.refreshItems = () => setRefreshCounter(prev => prev + 1);
 
-  useEffect(() => {
-    setInterval(() => {
-      setItems(prev => [...prev, ...generateItems(1)]);
-    }, 10000);
-  }, []);
+  useEffectOnUpdate(() => {
+    setItems(prev => [...prev.slice(1), ...generateItems(1)]);
```

Comment:
Not sure I understand this part?
Render 1: Renders with initial state of items. This hook does not run.
Render 2: Am just assuming there's something on this page which triggers a second render but that then trigger this hook and calls `setItems` which then triggers a new render
Render 3: Renders with rotated items

I guess if it does what is intended to be tested then its fine but was difficult for me to follow what the initial state of the page is meant to be.